iban/iban.go

```diff
@@ -143,3 +153,35 @@ func validate(value string) (bban.Structure, error) {
 	}
 	return struc, nil
 }
+
+func validateIgnoringTrailingGarbage(value string) (bban.Structure, error) {
+	if err := validateMinLength(value); err != nil {
+		return bban.Structure{}, err
+	}
+
+	code := extractCountryCode(value)
+	if err := validateCountryCode(code); err != nil {
+		return bban.Structure{}, err
+	}
+
+	struc, ok := country.GetBbanStructure(code)
+	if !ok {
+		return bban.Structure{}, ErrCountryCodeNotPresent
+	}
+
+	// Ignore anything beyond end of this country's IBAN format
+	properLength := struc.Length() + 4
+	if len(value) > properLength {
+		value = value[0:properLength]
+	}
```

Comment:
Seems like a lot to duplicate and maybe hard to keep in sync in the future. Any interest in moving this block to the private `validate()` function and using a function parameter to know when to use it? Like, `ignoreDataAfterIBAN` or something?